### DISCUSSION TITLE
View (Markdown) Source button on DEV

### DISCUSSION THREAD

**Discussion by szabgab** [forem/forem#18912] [Feature Requests]
One of the best ways to learn anything is to be able to look at the source code. Browsers provide an easy way to see the HTML source code of a web site.

I think it would be nice to have a button (maybe behind the ... on each article and each comment) that you could click and see the Markdown source of it.

For example I'd like to be able to see how the table of contents was added to [this article](https://dev.to/matusstafura/introduction-to-singly-linked-list-and-basic-operations-in-php-1d71).

I think this would be a nice way to help people improve their knowledge of Markdown

Originally posted [here](https://dev.to/szabgab/view-markdown-source-button-on-dev-19n8) where I already got an example for the TOC. The idea in general still stands.

**Comment by michael-tharrington** [comment]
This is a pretty awesome idea!

Like I said on DEV:
> I like the reasoning behind it. Though, it does make me slightly worried that it would make it even easier for plagiarists to do their thing. Might be worth it anyway.

Anyway, I'll return if I have more thoughts on this, but appreciate ya sharing!

**Comment by szabgab** [comment]
I noticed that if I copy-paste an article I am reading (so the already rendered version), the links will be in Markdown form. Is this done by some Forem-sepecific JavaScript code?

Answering myself: probably not as the same happens on the original post in this discussion, here on GitHub.
